### PR TITLE
Adds CI workflow using github actions

### PR DESCRIPTION
**Pull request by gruyaume**
- Adds CI workflow using github actions
- Fixes linting, static analysis
- Removes unit tests that failed anyway
- Improves README.md
- Creates CONTRIBUTING.md

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -149,19 +150,19 @@ def _on_install(self, _):
         service_enable(SRS_ENB_SERVICE)
         self._stored.installed = True
 
-    def _on_start(self, _):
+    def _on_start(self, _: StartEvent):
```
Comment (sanchezfdezjavier) [inline src/charm.py:153]: We might want to add docstrings in this file as we're doing in `src/utils.py`, I don't know if you plan to do it in another PR.

File: src/utils.py [hunk 2]
```diff
@@ -83,45 +96,54 @@ def is_ipv4(ip: str) -> bool:
...
-def ip_from_iface(subnet: str) -> str:
+def ip_from_iface(subnet: str) -> Optional[str]:
+    """Returns Ip address from a given subnet."""
     try:
         target_network = IPNetwork(subnet)
         networks = get_local_ipv4_networks()
         for network in networks:
             if network.ip in target_network:
                 return network.ip.format()
+        return None
```
Comment (sanchezfdezjavier) [inline src/utils.py:147]: Also nitpicky, still, cleaner and shorter?
```suggestion
        return next((network.ip.format() for network in networks if network.ip in target_network), None)
```

File: src/utils.py [hunk 3]
```diff
@@ -18,10 +24,12 @@ def service_active(service_name: str):
 
 
 def all_values_set(dictionary: Dict[str, str]) -> bool:
+    """Returns whether all values in a dict are set."""
     return not any(v is None for v in dictionary.values())
```
[the remark this answers is not in the document]
Comment (gruyaume) [reply src/utils.py:28]: I think I agree but it's out of scope of my PR here.

File: src/utils.py (hunk 2 above)
Comment (gruyaume) [reply src/utils.py:147]: I think I agree but it's out of scope of my PR here.

File: src/utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (gruyaume) [reply src/utils.py:28]: There are no unit tests in this repo (yet) so I didn't want to risk any change that could break anything.

File: src/charm.py (hunk 1 above)
Comment (gruyaume) [reply src/charm.py]: Done

File: src/utils.py (hunk 2 above)
Comment (sanchezfdezjavier) [reply src/utils.py:147]: Got it, makes sense

**Review by sanchezfdezjavier — APPROVED** [review]

**Review by ghislainbourgeois — APPROVED** [review]

File: charmcraft.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+type: "charm"
+bases:
+  - build-on:
+    - name: "ubuntu"
+      channel: "20.04"
+    - name: "ubuntu"
+      channel: "22.04"
```
Comment (sanchezfdezjavier) [inline charmcraft.yaml:10]: Shouldn't this be dropped?

Comment (gruyaume) [reply charmcraft.yaml:10]: How so? I took the same file content we had in our charms

**Review by markbeierl — APPROVED** [review]
